### PR TITLE
Feature/energy balance init

### PR DESCRIPTION
**Pull request by vgro**
This PR adds the files for the energy balance submodule and the setup() of variables that are required for the energy balance. The setup is copied from the simple abiotic model and there is an open to do to make the handling of constants neater, I would like to leave this for now as we are still moving constants around. I also deleted a few unneccessary bits here and there.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: virtual_rainforest/models/abiotic/__init__.py [hunk 1]
```diff
@@ -20,6 +20,10 @@
   above- and within-canopy wind profiles for the Virtual Rainforest. These profiles will
   determine the exchange of heat, water, and :math:`CO_{2}` between soil and atmosphere
   below the canopy as well as the exchange with the atmsophere above the canopy.
+
+* The :mod:`~virtual_rainforest.models.abiotic.energy_balance` submodule calculates the
+  energy balance of the Virtual Rainforest. The module returns vertical profiles of air
+  temperature, relative humididty, and soil temperature.
```
Comment (vgro) [inline virtual_rainforest/models/abiotic/__init__.py:26]:
```suggestion
  temperature, relative humidity, and soil temperature.
```

File: virtual_rainforest/models/abiotic/abiotic_model.py [hunk 2]
```diff
@@ -129,7 +122,36 @@ def from_config(
...
+            dims=["layers", "cell_id"],
+            coords={
+                "layers": np.arange(0, len(self.layer_roles)),
+                "layer_roles": ("layers", self.layer_roles),
+                "cell_id": self.data.grid.cell_id,
+            },
+            name="soil_temperature",
+        )
+
+        # Calculate vapour pressure deficit at reference height for all time steps
```
Comment (vgro) [inline virtual_rainforest/models/abiotic/abiotic_model.py:145]:
```suggestion
        # Calculate vapour pressure deficit at reference height for all time steps
        # TODO sort out constants argument in simple abiotic model
```

**Review by jacobcook1995 — APPROVED** [review]
LGTM!

**Comment by jacobcook1995** [comment]
Only thing I might suggest is making the open TODO into a Github issue, basically so that the issue is recorded somewhere that other people check

**Pull request merged**